app/src/main/java/org/keepgoeat/presentation/home/HomeMyGoalAdapter.kt

```diff
@@ -0,0 +1,109 @@
+package org.keepgoeat.presentation.home.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import org.keepgoeat.R
+import org.keepgoeat.databinding.ItemAddGoalBinding
+import org.keepgoeat.databinding.ItemMyGoalBinding
+import org.keepgoeat.presentation.home.MyGoalInfo
+import org.keepgoeat.presentation.type.HomeBtnType
+import org.keepgoeat.presentation.type.HomeGoalType
+import org.keepgoeat.util.setVisibility
+
+class HomeMyGoalAdapter : ListAdapter<MyGoalInfo, RecyclerView.ViewHolder>(
+    MyGoalDiffCallback
+) {
+    class MyGoalViewHolder(
+        private val binding: ItemMyGoalBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        var layout = binding
+        fun bind(myGoal: MyGoalInfo, goalType: HomeGoalType) {
+            val btnType: HomeBtnType = if (goalType == HomeGoalType.MORE) { // 더 먹기인 경우
+                if (myGoal.goalAchieved) {
+                    HomeBtnType.PLUS_ACHIEVED
+                } else {
+                    HomeBtnType.PLUS_NOT_ACHIEVED
+                }
+            } else { // 덜 먹기인 경우
+                if (myGoal.goalAchieved) {
+                    HomeBtnType.MINUS_ACHIEVED
+                } else {
+                    HomeBtnType.MINUS_NOT_ACHIEVED
+                }
+            }
+            binding.goal = myGoal
+            binding.goalType = goalType
+            binding.goalBtn = btnType
+        }
+    }
+
+    class AddGoalViewHolder(
+        private val binding: ItemAddGoalBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        var layout = binding
+        fun bind(goalCount: Int) {
+            when (goalCount) {
+                0 -> {}
+                1 -> binding.tvAddMoreGoal.setText(R.string.home_two_more_goal)
+                2 -> binding.tvAddMoreGoal.setText(R.string.home_one_more_goal)
+                3 -> {
+                    binding.tvAddMoreGoal.setText(R.string.home_no_more_goal)
+                    binding.btnMakeGoal.setVisibility(false)
+                }
+            }
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when (viewType) {
+            MyGoalInfo.MY_GOAL_TYPE -> {
+                val binding =
+                    ItemMyGoalBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+                MyGoalViewHolder(binding)
+            }
```

Comment:
```suggestion
    private lateinit var inflater: LayoutInflater
    
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        if (!::inflater.isInitialized)
                inflater = LayoutInflater.from(parent.context)
        return when (viewType) {
            MyGoalInfo.MY_GOAL_TYPE -> {
                val binding =
                    ItemMyGoalBinding.inflate(inflater, parent, false)
                MyGoalViewHolder(binding)
            }
```

이런식으로 매번 inflater를 만들지 않고 한번 만들어진 Inflater를 재사용하는 것이 좋아보입니다~